src/tests/conformance/invalid_message/disconnect.rs

```diff
@@ -193,63 +115,34 @@ async fn inv_with_mixed_types() {
 
 #[tokio::test]
 async fn addr_without_timestamp() {
-    // zcashd: pass
-    // zebra:  fail (replies with Reject(Malformed))
-
-    // Encode a custom type which mimics Message::Addr but without the timestamp.
-    let bytes = AddrWithoutTimestamp::new().encode().unwrap();
-
-    run_test_case_bytes(bytes).await.unwrap();
-}
-
-/// Mimics the encoding of [`NetworkAddr`] but excludes the timestamp field.
-struct NetAddrWithoutTimestamp(NetworkAddr);
-impl NetAddrWithoutTimestamp {
-    fn new() -> Self {
-        Self(NetworkAddr::new(SocketAddr::new(
-            IpAddr::V4(Ipv4Addr::new(0, 0, 0, 0)),
-            0,
-        )))
-    }
-}
-impl Codec for NetAddrWithoutTimestamp {
-    fn encode(&self, buffer: &mut Vec<u8>) -> io::Result<()> {
-        self.0.encode_without_timestamp(buffer)
-    }
-
-    fn decode(_bytes: &mut io::Cursor<&[u8]>) -> io::Result<Self>
-    where
-        Self: Sized,
-    {
-        unimplemented!("This is unused");
-    }
-}
-
-/// Mimics the encoding of a broken [`Message::Addr`].
-/// Contains a single [`NetAddrWithoutTimestamp`].
-///
-/// This is used by the [`addr_without_timestamp()`] test. [`Message::Addr`]
-/// cannot be used for this, as it does not support encoding without a timestamp (as
-/// this is obsolete behaviour).
-struct AddrWithoutTimestamp(Vec<NetAddrWithoutTimestamp>);
-impl AddrWithoutTimestamp {
-    fn new() -> Self {
-        Self(vec![NetAddrWithoutTimestamp::new()])
-    }
-
-    fn encode(&self) -> io::Result<Vec<u8>> {
-        let mut payload = Vec::new();
-        self.0.encode(&mut payload)?;
-
-        let header = MessageHeader::new(ADDR_COMMAND, &payload);
-
-        // Encode the header and append the message to it.
-        let mut buffer = Vec::with_capacity(HEADER_LEN + header.body_length as usize);
-        header.encode(&mut buffer)?;
-        buffer.append(&mut payload);
+    // zcashd: fail (replies with Reject(Malformed))
+    // zebra:  pass
 
-        Ok(buffer)
-    }
+    // Create a Addr message and encode it. This encoding includes the timestamp.
+    let message = Message::Addr(Addr::new(vec![NetworkAddr::new(SocketAddr::new(
+        IpAddr::V4(Ipv4Addr::new(1, 2, 3, 4)),
+        10,
+    ))]));
+    let mut payload = Vec::new();
+    message.encode(&mut payload).unwrap();
+
+    // Remove the timestamp bytes.
+    //
+    // First byte is a VarInt=1 containing the length of our NetworkAddr vector (1).
```

Comment:
Note: the number of bytes will change if the test is modified to use more addresses, perhaps something to make _extra_ clear here, as this is easy to overlook.

Reply:
I've opted to make the length calculation explicit. Let me know if you prefer this; or rather a comment?

The change does require adding a `pub` constructor for `VarInt`.